### PR TITLE
Add length validation for attributes

### PR DESCRIPTION
**Pull request by kayathiri4**
$subject

### CODE REVIEW

File: components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/listener/SCIMUserOperationListener.java [hunk 1]
```diff
@@ -266,6 +271,41 @@ private void validateClaimValueForRegex(String claimURI, String claimValue, Stri
...
+            return;
+        }
+        String minLength = claimProperties.get(MIN_LENGTH);
+        String maxLength = claimProperties.get(MAX_LENGTH);
+        boolean required = Boolean.parseBoolean(claimProperties.get(REQUIRED));
+
+        if (!required && StringUtils.isBlank(value)) {
+            return;
+        }
+        if ((StringUtils.isNotEmpty(minLength) && Integer.parseInt(minLength) > value.length()) ||
```
Comment (indeewari) [inline components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/listener/SCIMUserOperationListener.java:301]: Shall we use isNotBlank here ? What happens if there is a parsing exception due to invalid characters?

File: components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/listener/SCIMUserOperationListener.java [hunk 2]
```diff
@@ -266,6 +271,41 @@ private void validateClaimValueForRegex(String claimURI, String claimValue, Stri
...
+                (StringUtils.isNotEmpty(maxLength) && Integer.parseInt(maxLength) < value.length())) {
+            throw new UserStoreClientException(String.format(ERROR_CODE_LENGTH_VIOLATION.getDescription(),
+                    claimProperties.get(PROP_DISPLAYNAME), StringUtils.isNotEmpty(minLength) ? minLength : 0,
+                    StringUtils.isNotEmpty(maxLength) ? maxLength : 1024), ERROR_CODE_LENGTH_VIOLATION.getCode());
```
Comment (indeewari) [inline components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/listener/SCIMUserOperationListener.java:305]: What is the purpose of giving the threshold values in an exception?

File: components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/listener/SCIMUserOperationListener.java (hunk 1 above)
Comment (kayathiri4) [reply components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/listener/SCIMUserOperationListener.java]: Fixed with c47a013af43b8e50ada82cdecdf6a275bc31f2b0

File: components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/listener/SCIMUserOperationListener.java (hunk 2 above)
Comment (kayathiri4) [reply components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/listener/SCIMUserOperationListener.java:305]: If any of min or max length is configured, we do the validation and return the error. If either of them is not configured, we display the threshhold values.

Comment (indeewari) [reply components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/listener/SCIMUserOperationListener.java:305]: ok

**Review by indeewari — DISMISSED** [review]

**Pull request merged**